lib/shared/addon/components/container-shell/component.js

```diff
@@ -90,26 +85,30 @@ export default Ember.Component.extend(ThrottledResize, {
       });
       this.set('term', term);
 
-      term.on('data', function(data) {
-        //console.log('To Server:',data);
-        socket.send(AWS.util.base64.encode(data));
+      term.on('data', function (data) {
+        socket.send(`0${window.btoa(window.unescape(encodeURIComponent(data)))}`);
```

Comment:
Is there a reason for this hackery (and in the other places) instead of `AWS.util.base64.encode()` ?  `window.btoa` doesn't work with arbitrary binary data, the AWS one does.

Reply:
Updated